scenario_player/runner.py

```diff
@@ -153,279 +256,260 @@ def local_seed(self) -> str:
             seed = seed_file.read_text().strip()
         return seed
 
-    def select_chain(self, chain_urls: Dict[str, List[str]]) -> Tuple[str, List[str]]:
-        """Select a chain and return its name and RPC URL.
+    def ensure_token_network_discovery(self) -> ChecksumAddress:
```

Comment:
Corresponds to the old [`ensure_token_network_discovery`](https://github.com/raiden-network/scenario-player/blob/f208ed6cb4f713c1efc921b10458365bcf918b5d/scenario_player/runner.py#L228-L235). Improve it by making sure the nodes don't diverge (all report the token network address).